File: java/rest-simple-exercise/src/test/java/com/example/restsimple/StudentControllerTest.java
package com.example.restsimple;

import com.example.restsimple.controller.StudentController;
import com.example.restsimple.model.Student;
import com.example.restsimple.repository.UpdateStudent;
import com.example.restsimple.service.StudentService;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;

import java.util.Collections;
import java.util.Optional;
import java.util.UUID;

import static org.mockito.Mockito.*;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

public class StudentControllerTest {

    private StudentService studentService;
    private MockMvc mockMvc;

    @BeforeEach
    public void setUp() {
        studentService = mock(StudentService.class);
        StudentController studentController = new StudentController(studentService);
        mockMvc = MockMvcBuilders.standaloneSetup(studentController).build();
    }

    @Test
    public void testGetAllStudents() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testCreateStudent() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testUpdateStudent() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testDeleteStudent() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testGetAllStudents_EmptyList() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testCreateStudent_MissingName() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testCreateStudent_MissingLastName() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testUpdateStudent_NotFound() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Test
    public void testDeleteStudent_NotFound() throws Exception {
        throw new UnsupportedOperationException("Not implemented");
    }
}

File: kotlin/rest-simple-exercise/src/test/kotlin/com/example/restsimple/StudentControllerTest.kt
package com.example.restsimple

import com.example.restsimple.model.Student
import com.example.restsimple.repository.UpdateStudent
import com.example.restsimple.request.StudentCreateRequest
import com.example.restsimple.request.StudentUpdateRequest
import com.example.restsimple.service.StudentService
import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Test
import org.mockito.kotlin.anyOrNull
import org.mockito.kotlin.whenever
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
import java.time.LocalDateTime
import java.util.*

@SpringBootTest
@AutoConfigureMockMvc
internal class StudentControllerTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

    @MockBean
    private lateinit var studentService: StudentService

    @Autowired
    private lateinit var objectMapper: ObjectMapper

    @Test
    fun `test health endpoint`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test get all students`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test create student`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test successful student update`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test update student not found`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test delete student`() {
        throw UnsupportedOperationException("Not implemented");
    }

    @Test
    fun `test delete student not found`() {
        throw UnsupportedOperationException("Not implemented");
    }
}

